packages/react-components/src/components/gauge/utils/getThresholds.ts

```diff
@@ -1,29 +1,113 @@
+  settings?: GaugeSettings
+) => {
+  const lineStylesGTandGTE: [number, string][] = [];
+  let lastThreshold: ThresholdRange | undefined = undefined;
+
+  // filter thresholds to only be GT or GTE, then sort them from greatest to smallest
+  const gteThresholds = allThresholds
+    .filter(
+      (tr) => tr.comparisonOperator === 'GT' || tr.comparisonOperator === 'GTE'
+    )
+    .sort((a, b) => (b.start ?? 0) - (a.start ?? 0));
+
+  // since the thresholds are sorted in descending order, the first one we see will provide the end range for the second threshold
+  // basically each preceding threshold provides the ceiling for the next one
+  gteThresholds.forEach((thresholdRange) => {
+    const { start, color } = thresholdRange;
+    if (!lastThreshold) {
+      start && lineStylesGTandGTE.push([1, color]);
+    } else {
+      lastThreshold.start &&
+        lineStylesGTandGTE.push([lastThreshold.start, color]);
+    }
+    lastThreshold = thresholdRange;
+  });
+
+  // once all thresholds have been calculated, the remaining space has to be filled in with the default gauge color
+  if (lastThreshold) {
+    lastThreshold['start'] &&
+      lineStylesGTandGTE.push([
+        lastThreshold['start'],
+        settings?.color ?? DEFAULT_GAUGE_PROGRESS_COLOR,
+      ]);
+  }
+
+  return lineStylesGTandGTE;
+};
+
+// convert thresholds from our definition to a [number, pair] format which echarts will use to render the correct colors and labels
 export const getThresholds = ({
   settings,
   thresholds,
 }: Pick<GaugeProps, 'settings' | 'thresholds'>) => {
-  const newArray = thresholds?.map(({ value, color }) => {
-    return [
-      getDecimalValueFromMinMax({
-        value,
-        min: settings?.yMin,
-        max: settings?.yMax,
-      }),
-      color,
-    ];
-  });
+  // create a list of thresholds with Start and End ranges (depending on the comparison operator)
+  const thresholdsWithRanges: ThresholdRange[] =
+    thresholds
+      ?.map(({ value, color, comparisonOperator }) => {
+        const [start, end] = getThresholdRangeFromMinMax({
+          value,
+          comparisonOperator,
+          min: settings?.yMin,
+          max: settings?.yMax,
+        });
 
-  /*
-   * Add default threshold at 100% if none is defined
-   * This is required for ECharts to show the gauge correctly
-   */
-  if (!newArray?.some((item) => item[0] === 1)) {
-    newArray?.push([1, DEFAULT_GAUGE_PROGRESS_COLOR]);
-  }
+        return {
+          start,
+          end,
+          color,
+          comparisonOperator,
+        };
+      })
+      .filter((tr) => tr.start !== tr.end) ?? [];
+
+  // LT LTE
+  const colorValuePairsLTandLTE =
+    getColorValuePairsForLTandLTE(thresholdsWithRanges);
+
+  // GT GTE
+  const colorValuePairsGTandGTE = getColorValuePairsForGTandGTE(
+    thresholdsWithRanges,
+    settings
+  );
+
+  // EQ and CONTAINS are not supported since they do not make sense in the case of a gauge widget
```

Comment:
Is the necessary? Could this be explained with code? Such as a "SupportedOperators" type?